### PR TITLE
[Backend Issue #5]: Add database fields and update necessary get/set functions and enumerators

### LINKED ISSUE #5: [Backend]: Add to database to hold necessary document data
**How to tackle this?**
What needs to be changed about the backend is that we need to alter the current database such that it holds all of our necessary document data.
Database needs to be able to hold important data about the document, namely the status, the priority, and the last modified date.
The database has a current location in the backend to store documents itself, so as a sub-task, we need to find that and see if we can associate these new fields with that field.

**Dependencies:**
Middleware: The main middleware dependency here is that backend relies on middleware to retrieve the correct data from the database from the location, as well as read into the data structure that it might be stored in properly.
Middleware is also responsible for making sure the data being sent in is sorted
Frontend: Backend is pretty reliable on frontend, as frontend is responsible for displaying all the info stored in backend.

**Acceptance Criteria:**
Testing to make sure that all the necessary data is stored with the sorting that middleware gives it.
Need to make sure that data in the database is being changed when frontend and middleware is changing it.

**Time Estimation:**
Locating areas that need to be changed -> 1 day (Saturday)
Change necessary parts of database and database access ->1-2 days (Sunday-Monday)
Depending on the number of files, this is a variable time period. This could extend beyond Monday, in which office hours would be prioritized.
Deadline for backend:  Monday afternoon (6:30pm) -> Tuesday afternoon at the latest, on the occasion office hours is a necessity

### PR DESCRIPTION
**Pull request by NRAM3SH**
Added fields in both route steps and document database tables to include priority and status fields. I also updated the necessary get and set fucntions in Document, DocumentDTO and RouteStep ot receive this data. This involved creating new enumerators for these route step types, as seen in the created RouteStepPriority and RouteStepStatus files. Resolve issue #5.

### CODE REVIEW

File: docs-core/src/main/java/com/sismics/docs/core/model/jpa/Document.java [hunk 1]
```diff
@@ -257,6 +269,22 @@ public void setUpdateDate(Date updateDate) {
...
+    public String getPriority() {
+        return priority;
+    }
+
+    public void setPriority(String priority) {
+        this.priority = priority;
+    }
+
+    public String getStatus() {
+        return priority;
```
Comment (SapheSab3r) [inline docs-core/src/main/java/com/sismics/docs/core/model/jpa/Document.java]: shouldn't this be `return status`?

**Comment by jackwu999** [comment]
Please add line `q.setParameter("status", RouteStepStatusType.valueOf("COMPLETE"));` in file `RouteStepDao.java` under `endRouteStep` method. This changes the status to complete after we validate the route step

**Review by mebaker520 — APPROVED** [review]
looks good to merge

**Review by jackwu999 — APPROVED** [review]

**Review by SapheSab3r — APPROVED** [review]
All fixed. I approved.

**Comment by SapheSab3r** [comment]
UPDATE: the type is not String for all the variables you added

**Pull request merged**